src/main/java/com/vadmack/mongodbtest/controller/AuthController.java

```diff
@@ -0,0 +1,53 @@
+package com.vadmack.mongodbtest.controller;
+
+import com.vadmack.mongodbtest.dto.AuthRequest;
+import com.vadmack.mongodbtest.dto.UserDto;
+import com.vadmack.mongodbtest.entity.User;
+import com.vadmack.mongodbtest.security.JwtTokenUtil;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(path = "/api/public")
+@RequiredArgsConstructor
+public class AuthController {
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenUtil jwtTokenUtil;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @PostMapping("/login")
+    public ResponseEntity<UserDto> login(@RequestBody @Valid AuthRequest request) {
+        try {
+            Authentication authenticate = authenticationManager
+                    .authenticate(
+                            new UsernamePasswordAuthenticationToken(
+                                    request.getUsername(), request.getPassword()
+                            )
+                    );
+
+            User user = (User) authenticate.getPrincipal();
+
+            return ResponseEntity.ok()
+                    .header(
+                            HttpHeaders.AUTHORIZATION,
+                            jwtTokenUtil.generateAccessToken(user)
+                    )
+                    .body(modelMapper.map(user, UserDto.class));
+        } catch (BadCredentialsException ex) {
```

Comment:
Maybe it would be better to handle the exception using `CustomExceptionHandler`?